File: tell-dont-ask-kata-kotlin/src/test/java/it/gabrieletondi/telldontaskkata/doubles/InMemoryProductCatalog.kt
package test.java.it.gabrieletondi.telldontaskkata.doubles


import main.java.it.gabrieletondi.telldontaskkata.domain.Product
import main.java.it.gabrieletondi.telldontaskkata.repository.ProductCatalog

class InMemoryProductCatalog(private val products: List<Product>) : ProductCatalog {

    override fun getByName(name: String): Product? {
        return products.stream().filter { p -> p.name.equals(name) }.findFirst().orElse(null)
    }
}

File: tell-dont-ask-kata-csharp/TellDontAsk.Tests/doubles/InMemoryProductCatalog.cs
using System.Collections.Generic;
using System.Linq;

public class InMemoryProductCatalog : ProductCatalog
{
    private readonly List<Product> products;

    public InMemoryProductCatalog(List<Product> products)
    {
        this.products = products;
    }

    public Product GetByName(string name)
    {
        return products.FirstOrDefault(p => p.GetName().Equals(name));
    }
}